src/codegen.cpp

```diff
@@ -4825,7 +4856,18 @@ static std::unique_ptr<Module> emit_function(
     if (va && ctx.vaSlot != -1) {
         jl_varinfo_t &varinfo = ctx.slots[ctx.vaSlot];
         varinfo.isArgument = true;
-        varinfo.value = mark_julia_type(ctx, (Value*)NULL, false, jl_tuple_type);
+        if (specsig) {
+            size_t nvargs = jl_nparams(lam->specTypes)-nreq;
+            jl_svec_t *tupargs = jl_alloc_svec(nvargs);
```

Comment:
missing gc root

Reply:
also might be good to pull this out into a function that computes the type of vararg